subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AnnotationProcessorDetector.java

```diff
@@ -47,6 +49,7 @@
  * Inspects a classpath to find annotation processors contained in it. If several versions of the same annotation processor are found,
  * the first one is returned, mimicking the behavior of {@link java.util.ServiceLoader}.
  */
+@ServiceScope(Scopes.BuildTree)
```

Comment:
@melix From my understanding, this change is not an issue.
Can you confirm?

Reply:
@melix ping

Reply:
The annotation itself does nothing, if that's what you ask. 

If you did the change to the actual creation, you're moving a service from "Gradle scope" to "build tree scope", which means the caches it contains will not survive cross builds. It's likely to have an impact on performance and memory consumption. Why is this change needed?

Reply:
Yes, the annotation is only used by the configuration cache. The driver was configuration cache to move those services. Running the integration tests with configuration cache enabled fails as cc only supports build tree services at this point. @eskatos suggested to try and move the services into tree scope (if possible). If not, there is another workaround to consume the services (by registering them in `Codecs.kt`).
How could I find out what kind of performance impact this has?
I just tried to only move the other services into build tree, keeping the `AnnotationProcessorDetector` in `Gradle` and add a `ownerServiceCodec` for that but this doesn't seem to work (maybe I'm doing something wrong?)
```
            > Cannot create service of type JavaCompilerFactory using method JavaToolChainServiceRegistry$.createJavaCompilerFactory() as required service of type AnnotationProcessorDetector for parameter #6 is not available.
```

Reply:
Pushed another branch with those changes: https://github.com/gradle/gradle/commit/0925acde226b75cafbb6e1384f7bfa439015f8b3

`JavaCompileToolchainIntegrationTest` is the relevant test for this

Reply:
See discussion on PR